### PR TITLE
[Rest Api Compatibility] Types for Percolate Query Api

### PR DESCRIPTION
**Pull request by pgomulka**
Previously removed in #46985. The yaml test is included in this PR, but
will be removed once #74689 is merged.

relates #54160 
relates main meta issue #51816

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine retest this please

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/docs

File: modules/percolator/build.gradle [hunk 1]
```diff
@@ -8,6 +8,7 @@
 apply plugin: 'elasticsearch.yaml-rest-test'
 apply plugin: 'elasticsearch.yaml-rest-compat-test'
 apply plugin: 'elasticsearch.internal-cluster-test'
+apply plugin: 'elasticsearch.yaml-rest-compat-test'
```
Comment (romseygeek) [inline modules/percolator/build.gradle:11]: This is a duplicate of line 9

**Review by romseygeek — APPROVED** [review]
LGTM

**Pull request merged**